packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4148,6 +4148,43 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       : TextPosition(offset: textBoundary.getLeadingTextBoundaryAt(caretOffset) ?? 0);
   }
 
+  /// Select a paragraph around the given position.
+  ///
+  /// A paragraph boundary is defined as the range that most closely
+  /// encapsulates the given position within two line terminators. If a
+  /// line terminator does not exist in a given direction the selection
+  /// extends to the start/end of the document in that direction.
+  void selectParagraph({required Offset position, SelectionChangedCause? cause}) {
+    selectParagraphsInRange(from: position, cause: cause);
```

Comment:
I think I want to avoid adding more methods to `RenderEditable` that mutate the selection, and instead have it provide selection calculations that `EditableText` uses to mutate its selection.

Reply:
For design clarity they should all be in a single place I think. It makes sense to move the selection modifying methods to `EditableText` but it looks like they were in `RenderEditable` for years so it'd be a breaking change. Wondering if @LongCatIsLooong or @justinmc may have some thoughts?

Reply:
I really like the idea of separating the calculation of text related stuff from its mutation.  I hope we can get to where we have a clear place for each.  For example, if some Flutter dev wants to create a button that selects the nearest 3 paragraphs or something, it would be nice to be able to give them access to our paragraph selection logic without making them actually make a change to an EditableText.

For now though it looks like Renzo has made selectParagraphsInRange and selectParagraph private and moved them to text_selection.dart.  That's definitely fine for now since we can always change it later.